packages/date/src/createInlineDatePlugin.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  createPluginFactory,
+  getNodeEntry,
+  getRange,
+  isEndPoint,
+  isSelectionExpanded,
+  isStartPoint,
+} from '@udecode/plate-common/server';
+import { Path } from 'slate';
+
+export const ELEMENT_INLINE_DATE = 'inline_date';
+
+export const createInlineDatePlugin = createPluginFactory({
+  isElement: true,
+  isInline: true,
+  isVoid: true,
+  key: ELEMENT_INLINE_DATE,
+  withOverrides: (editor) => {
```

Comment:
I can't tell from reading the code what this override does. If it's meant to prevent the element from being selected using the arrow keys, you should be able to achieve this more easily by overriding `editor.isSelectable` to return false for this element.

If this override is doing something besides that, can you add comments to explain?

Reply:
Oh, I just learned about this API thanks. However, I tried it, and it seems to move an extra point? It might be a bug in Slate.

text<cursor /><hinlinedate />text  
presss right arrow will be like this: 
text<hinlinedate>t<cursor>ext

Reply:
Can you reproduce that issue on https://www.slatejs.org/examples/inlines if you copy and paste the green "Approved" badge?

Reply:
Cannot reproduce on Slate because of slate do it.
https://github.com/ianstormtaylor/slate/blob/main/site/examples/inlines.tsx#L74-L96

 I can’t do the same thing in handler.onKeyDown to solve this problem.
 But I can rewrite move in `withOverrides` to solve.
`   editor.move = (op) => {
      if (editor.selection && isCollapsed(editor.selection)) {
        return move({
          ...op,
          unit: 'offset',
        });
      }

      move(op);
    };`

Reply:
Changing the unit for `move` throughout the whole editor is quite a big side effect of enabling the date plugin, and may cause issues. Maybe only override this behaviour if the cursor is next to a date node?

Reply:
emm then the code is back to its original.The original code wasn’t written in vain; it’s coincidentally being used again haha.

Reply:
Indeed. Can you refactor a helper function or two out of the override so that the override is more focused on _what_ it's doing rather than _how_ it's doing it?

Reply:
For instance, all of the logic for checking whether the cursor is next to a node of a certain type could be refactored out.

Reply:
Sure. thank for your advise.